src/definitions/x509/trust_anchor.rs

```diff
@@ -0,0 +1,379 @@
+        return Err(X509Error::ValidationError(
+            "The signer certificate is expected to have extensions, but none were found. Skipping all following extension validation checks.. "
+                .to_string(),
+        ));
+    };
+
+    match rule_set.typ {
+        //Under the IACA ruleset, the values for S or ST should be the same in subject and issuer if they are present in both
+        RuleSetType::IACA => {
+            let mut extension_errors = validate_iaca_root_extensions(root_extensions);
+            extension_errors.append(&mut validate_iaca_signer_extensions(
+                leaf_extensions,
+                MDOC_VALUE_EXTENDED_KEY_USAGE,
+            ));
+            for dn in leaf_distinguished_names {
+                if dn.oid.to_string() == *"2.5.4.8" {
+                    let state_or_province =
+                        root_distinguished_names.iter().find(|r| r.oid == dn.oid);
+                    if let Some(st_or_s) = state_or_province {
+                        if dn != *st_or_s {
+                            return Err(X509Error::ValidationError(format!("Mismatch between supplied certificate issuer attribute: {:?} and the trust anchor registry.", dn)));
+                        }
+                    }
+                } else {
+                    let Some(_root_dn) = root_distinguished_names.iter().find(|r| r == &&dn) else {
+                        return Err(X509Error::ValidationError(format!("Mismatch between supplied certificate issuer attribute: {:?} and the trust anchor registry.", dn.value)));
+                    };
+                }
+            }
+            Ok(extension_errors)
+        }
+        //Under the AAMVA ruleset, S/ST is mandatory and should be the same in the subject and issuer
+        RuleSetType::AAMVA => {
+            let mut extension_errors = validate_iaca_root_extensions(root_extensions);
+            extension_errors.append(&mut validate_iaca_signer_extensions(
+                leaf_extensions,
+                MDOC_VALUE_EXTENDED_KEY_USAGE,
+            ));
+            for dn in leaf_distinguished_names {
+                let Some(_root_dn) = root_distinguished_names.iter().find(|r| r == &&dn) else {
+                    return Err(X509Error::ValidationError(format!("Mismatch between supplied certificate issuer attribute: {:?} and the trust anchor registry.", dn.value)));
+                };
+            }
+            Ok(extension_errors)
+        }
+        RuleSetType::NamesOnly => {
+            for dn in leaf_distinguished_names {
+                let Some(_root_dn) = root_distinguished_names.iter().find(|r| r == &&dn) else {
+                    return Err(X509Error::ValidationError(format!("Mismatch between supplied certificate issuer attribute: {:?} and the trust anchor registry.", dn.value)));
+                };
+            }
+            Ok(vec![])
+        }
+        RuleSetType::ReaderAuth => {
+            //TODO
```

Comment:
@justAnIdentity do you remember what needs to be done here?

Reply:
This should follow the certificate profile as defined on pages 78-80 of 18013-5 (mdoc reader authentication). It should be pretty similar to IACA, but there specifically is a value for extended key usage that should be **1.0.1803.5.6**